libsolidity/parsing/Scanner.cpp

```diff
@@ -768,8 +768,14 @@ Token::Value Scanner::scanNumber(char _charSeen)
 			scanDecimalDigits();  // optional
 			if (m_char == '.')
 			{
+				// A '.' has to be followed by a number.
+				if (!isDecimalDigit(m_source.get(1)))
```

Comment:
I think this should also use `m_source.isPastEndOfInput`. To ensure it is tested, please add a test case of `contract C { uint a = 2.`